# Copyright (c) 2007, 2009, Oracle and/or its affiliates. All rights reserved.
#
# Permission is hereby granted, free of charge, to any person obtaining a
# copy of this software and associated documentation files (the "Software"),
# to deal in the Software without restriction, including without limitation
# the rights to use, copy, modify, merge, publish, distribute, sublicense,
# and/or sell copies of the Software, and to permit persons to whom the
# Software is furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice (including the next
# paragraph) shall be included in all copies or substantial portions of the
# Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
# DEALINGS IN THE SOFTWARE.
Make docbook2txt work on Solaris for Xorg doc builds

diff -urp -x '*~' -x '*.orig' backends/man.in backends/man.in
--- backends/man.in	2003-02-11 04:56:23.000000000 -0800
+++ backends/man.in	2009-07-12 10:22:31.006361770 -0700
@@ -3,7 +3,7 @@
 # This program is under GPL license. See LICENSE file for details.
 
 # Convert to *roff
-HELPER=$SGML_BASE_DIR/docbook/utils-@VERSION@/helpers/docbook2man-spec.pl
+HELPER=@prefix@/share/sgml/docbook/utils-@VERSION@/helpers/docbook2man-spec.pl
 TMPDIR=`mktemp -d /tmp/man.XXXXXX` || \
  { echo >&2 "man backend: could not create secure temporary directory"; exit 1;}
 trap 'rm -rf "${TMPDIR}"' EXIT
diff -urp -x '*~' -x '*.orig' backends/txt backends/txt
--- backends/txt	2004-02-11 05:58:03.000000000 -0800
+++ backends/txt	2009-07-12 10:22:31.006528202 -0700
@@ -1,3 +1,4 @@
+#! /usr/bin/bash
 # Backend to convert something into ASCII text
 # Send any comments to Eric Bischoff <eric@caldera.de>
 # This program is under GPL license. See LICENSE file for details.
@@ -6,6 +7,10 @@ if [ -x /usr/bin/lynx ]
 then
   CONVERT=/usr/bin/lynx
   ARGS="-force_html -dump -nolist -width=72"
+elif [ -x /opt/sfw/bin/lynx ]
+then
+  CONVERT=/opt/sfw/bin/lynx
+  ARGS="-force_html -dump -nolist -width=72"
 elif [ -x /usr/bin/links ]
 then
   CONVERT=/usr/bin/links
diff -urp -x '*~' -x '*.orig' bin/jw.in bin/jw.in
--- bin/jw.in	2003-04-30 09:21:49.000000000 -0700
+++ bin/jw.in	2009-07-12 10:22:31.006831201 -0700
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! /bin/bash
 # Jade Wrapper
 # Script to convert a SGML file into some other format
 # Send any comments to Eric Bischoff <eric@caldera.de>
@@ -86,11 +86,13 @@ then
   SGML_CATALOGS_DIR=`grep $RE $SGML_CONF | sed "s/$RE//"`
 fi
 
+DOCBOOK_UTILS_DIRS="@prefix@/share/sgml/docbook/utils-@VERSION@"
+
 # Set frontend to use
-SGML_FRONTEND="$SGML_BASE_DIR/docbook/utils-@VERSION@/frontends/docbook"
+SGML_FRONTEND="$DOCBOOK_UTILS_DIRS/frontends/docbook"
 
 # Set backend to use
-SGML_BACKEND="$SGML_BASE_DIR/docbook/utils-@VERSION@/backends/html"
+SGML_BACKEND="$DOCBOOK_UTILS_DIRS/backends/html"
 
 # Set main stylesheet
 SGML_STYLESHEET="default"
@@ -120,14 +122,14 @@ while [ $# -gt 0 ]
 do case $1 in
   -f|--frontend) case "$2" in
 		  /*) SGML_FRONTEND="$2" ;;
-		   *) d=$SGML_BASE_DIR/docbook/utils-@VERSION@/frontends
+		   *) d=$DOCBOOK_UTILS_DIRS/frontends
 		      SGML_FRONTEND="$d/$2" ;;
 		esac
 		shift 2
 		;;
   -b|--backend)	case "$2" in
 		  /*) SGML_BACKEND="$2" ;;
-		   *) d=$SGML_BASE_DIR/docbook/utils-@VERSION@/backends
+		   *) d=$DOCBOOK_UTILS_DIRS/backends
 		      SGML_BACKEND="$d/$2" ;;
 		esac
 		shift 2
@@ -306,19 +308,19 @@ fi
 # Set path to SGML catalogs (first try centralized catalog)
 case $SGML_STANDARD_CATALOGS in
   yes)	export SGML_CATALOGS_DIR SGML_FILE SGML_XML SGML_NORM
-	SGML_CENTRALIZED_CATALOG=`sh $SGML_FRONTEND centralized-catalog`
+	SGML_CENTRALIZED_CATALOG=`bash $SGML_FRONTEND centralized-catalog`
 	if [ -s $SGML_CENTRALIZED_CATALOG ]
 	then
 	  SGML_CATALOG_FILES=$SGML_CENTRALIZED_CATALOG
 	else
 	  SGML_CATALOG_FILES=`find $SGML_BASE_DIR -name catalog`
-	  SGML_CATALOG_FILES=`echo "$SGML_CATALOG_FILES" | tr [:space:] :`
+	  SGML_CATALOG_FILES=`echo "$SGML_CATALOG_FILES" | tr '[:space:]' :::::::`
 	fi
 	;;
   no)	SGML_CATALOG_FILES=""
 	;;
 esac
-SGML_CATALOG_FILES=${SGML_CATALOG_FILES}${SGML_EXTRA_CATALOGS}
+SGML_CATALOG_FILES=@prefix@/share/sgml/catalog:${SGML_CATALOG_FILES}${SGML_EXTRA_CATALOGS}
 SGML_CATALOG_FILES=`echo "$SGML_CATALOG_FILES" | sed 's/^://;s/:$//'`
 if [ -z "$SGML_CATALOG_FILES" ]
 then
@@ -340,7 +342,7 @@ esac
 if [ "$SGML_STYLESHEET" = "default" -o "$SGML_STYLESHEET" = "none" ]
 then
   export SGML_BASE_DIR SGML_TYPE SGML_STYLESHEET
-  SGML_STYLESHEET=`sh $SGML_FRONTEND style-sheet`
+  SGML_STYLESHEET=`bash $SGML_FRONTEND style-sheet`
   SGML_RETURN=$?
   if [ $SGML_RETURN -gt 0 ]
   then exit `expr 7 + $SGML_RETURN`
@@ -428,9 +430,9 @@ export SGML_CATALOG_FILES SGML_BASE_DIR 
 NOCHUNKS=`echo $SGML_OPTIONS | grep nochunks`
 if [ -z "$NOCHUNKS" ]
 then
-	sh $SGML_BACKEND
+	bash $SGML_BACKEND
 else
-	sh $SGML_BACKEND >$SGML_FILE_NAME.html
+	bash $SGML_BACKEND >$SGML_FILE_NAME.html
 fi
 SGML_RETURN=$?
 cd $SGML_CURRENT_DIRECTORY
diff -urp -x '*~' -x '*.orig' doc/HTML/Makefile.am doc/HTML/Makefile.am
--- doc/HTML/Makefile.am	2002-04-22 04:48:20.000000000 -0700
+++ doc/HTML/Makefile.am	2009-07-12 10:22:31.007235908 -0700
@@ -23,7 +23,7 @@ $(html_DATA): $(top_srcdir)/doc/docbook-
 		$(top_srcdir)/doc/refentry/docbook2texi-spec.pl.sgml \
 		$(top_srcdir)/doc/refentry/frontend-spec.sgml \
 		$(top_srcdir)/doc/refentry/sgmldiff.sgml
-	SGML_CATALOG_FILES=/etc/sgml/catalog \
+	SGML_CATALOG_FILES=/etc/sgml/catalog:$(top_srcdir)/catalog \
 	SGML_SEARCH_PATH=$(top_srcdir):$(top_srcdir)/doc:.. \
 		jade -t sgml -i html -d $(top_srcdir)/docbook-utils.dsl\#html \
 			-V '%use-id-as-filename%' $<
diff -urp -x '*~' -x '*.orig' doc/man/Makefile.am doc/man/Makefile.am
--- doc/man/Makefile.am	2002-06-27 04:37:07.000000000 -0700
+++ doc/man/Makefile.am	2009-07-12 10:22:31.007461556 -0700
@@ -10,7 +10,7 @@ CLEANFILES=$(man1_MANS) $(man7_MANS) man
 
 $(man1_MANS) $(man7_MANS): $(top_srcdir)/doc/docbook-utils.sgml \
 		$(top_srcdir)/helpers/docbook2man-spec.pl
-	SGML_CATALOG_FILES=/etc/sgml/catalog \
+	SGML_CATALOG_FILES=/etc/sgml/catalog:$(top_srcdir)/catalog \
 	SGML_SEARCH_PATH=$(top_srcdir)/doc:.. \
 		nsgmls $< | \
 		sgmlspl $(top_srcdir)/helpers/docbook2man-spec.pl
diff -urp -x '*~' -x '*.orig' frontends/docbook.in frontends/docbook.in
--- frontends/docbook.in	2002-11-02 09:45:10.000000000 -0800
+++ frontends/docbook.in	2009-07-12 10:22:31.006995792 -0700
@@ -15,7 +15,7 @@ case "$1" in
 	SGML_VERSION=`"$SGML_NORM" $SGML_NORM_OPTS -- "$SGML_FILE" 2>/dev/null |
 			grep -i '<!DOCTYPE' |
 			head -n 1 |
-			sed 's/^.*DocBook\( XML\)\? V\([0-9][\.0-9]*\).*$/\2/'`
+			perl -p -e 's/^.*DocBook\( XML\)\? V\([0-9][\.0-9]*\).*$/\2/'`
 	SGML_CATALOG=
 	for f in "${SGML_CATALOGS_DIR}/${SGML_XML}-docbook-${SGML_VERSION}"*".cat"
 	do 
@@ -38,10 +38,10 @@ case "$1" in
   style-sheet)
 	case $SGML_STYLESHEET in
 	  default)
-	    echo "${SGML_BASE_DIR}/docbook/utils-@VERSION@/docbook-utils.dsl#${SGML_TYPE}"
+	    echo "@prefix@/share/sgml/docbook/utils-@VERSION@/docbook-utils.dsl#${SGML_TYPE}"
 	    ;;
 	  none)
-	    find $SGML_BASE_DIR -name docbook.dsl | grep "$SGML_TYPE/docbook.dsl" | awk '{print $1}'
+	    find /usr/share/sgml -name docbook.dsl | grep "$SGML_TYPE/docbook.dsl" | awk '{print $1}'
 	    ;;
 	esac
         ;;
